modules/users/client/components/TopNavigationSmall.component.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import RemoveContact from '@/modules/contacts/client/components/RemoveContactContainer';
+
+export default function TopNavigationSmall({
+  username,
+  contact,
+  referencesEnabled,
+  isResolved,
+  selfId,
+  userId,
+  onContactRemoved,
+}) {
+  // @TODO this hacky fix should be removed and the data should be consistent
```

Comment:
This could warrant opening an issue as a follow up for some day.

Reply:
Maybe a reminder to review all those migration-dependent TODOs... seems there will be more as we progress.

I love Typescript, because it helps so much with fixing inconsistencies like this one...